engine/src/main/java/org/terasology/engine/world/generator/plugin/WorldGeneratorPluginManager.java

```diff
@@ -0,0 +1,87 @@
+
+    private final Context context;
+
+    public WorldGeneratorPluginManager(Context context) {
+        this.context = context;
+    }
+
+    /**
+     * @param urn urn of the world generator to create.
+     * @param context objects from this context will be injected into the world generator plugin
+     * @return The instantiated world generator plugin.
+     */
+    public static WorldGeneratorPlugin createGeneratorPlugin(ResourceUrn urn, Context context) throws UnresolvedWorldGeneratorPluginException {
+        ModuleManager moduleManager = context.get(ModuleManager.class);
+        Module module = moduleManager.getEnvironment().get(urn.getModuleName());
+        if (module == null) {
+            DependencyResolver resolver = new DependencyResolver(moduleManager.getRegistry());
+            ResolutionResult result = resolver.resolve(urn.getModuleName());
+            if (!result.isSuccess()) {
+                if (moduleManager.getRegistry().getLatestModuleVersion(urn.getModuleName()) == null) {
+                    throw new UnresolvedWorldGeneratorPluginException("Unable to resolve world generator plugin'"
+                            + urn + "' - not found");
+                } else {
+                    throw new UnresolvedWorldGeneratorPluginException("Unable to resolve world generator plugin'"
+                            + urn + "' - unable to resolve module dependencies");
+                }
+            }
+            try (ModuleEnvironment environment = moduleManager.loadEnvironment(result.getModules(), false)) {
+                return createWorldGeneratorPlugin(urn, context, environment);
+            }
+        } else {
+            return createWorldGeneratorPlugin(urn, context, moduleManager.getEnvironment());
+        }
+    }
+
+    /**
+     * @param urn urn of the world generator plugin to create.
+     * @param context that will be used to inject the world generator plugin.
+     * @param environment to be searched for the world generator plugin class.
+     * @return a new world generator plugin with the specified urn.
+     */
+    public static WorldGeneratorPlugin createWorldGeneratorPlugin(ResourceUrn urn, Context context, ModuleEnvironment environment)
+            throws UnresolvedWorldGeneratorPluginException {
+        for (Class<?> entry : environment.getTypesAnnotatedWith(RegisterPlugin.class)) {
+            ResourceUrn generatorUrn = new ResourceUrn(environment.getModuleProviding(entry).toString(), entry.getSimpleName());
+            if (generatorUrn.equals(urn)) {
+                WorldGeneratorPlugin worldGeneratorPlugin = loadGeneratorPlugin(entry, generatorUrn);
+                InjectionHelper.inject(worldGeneratorPlugin, context);
+                return worldGeneratorPlugin;
+            }
+        }
+        throw new UnresolvedWorldGeneratorPluginException("Unable to resolve world generator '" + urn + "' - not found");
+    }
```

Comment:
For the `createWorldGeneratorPlugin` method, I was not able to use the same code as in the `WorldGeneratorManager` as the `RegisterPlugin` does not have an `id` field like the `RegisterWorldGenerator` annotation.
https://github.com/MovingBlocks/Terasology/blob/ae83461fe1f6bca1b1c4706b976e49d38286066a/engine/src/main/java/org/terasology/engine/world/generator/internal/WorldGeneratorManager.java#L130-L145

I inspired myself from the `DefaultWorldGeneratorPluginLibrary` and needed to discuss if this was the right choice : 
https://github.com/MovingBlocks/Terasology/blob/ae83461fe1f6bca1b1c4706b976e49d38286066a/engine/src/main/java/org/terasology/engine/world/generator/plugin/DefaultWorldGeneratorPluginLibrary.java#L21-L29

Another option would have been to add the attributes to the `RegisterPlugin` annotation. Inconvenience would be that we would need an update to modules using the `RegisterPlugin` annotation to add this information. 

The advantage since the code is exactly the same, we could ponder on creating a generic class for this for which the inputs would be on one side `RegisterWorldGenerator/WorldGenerator` and on the other `RegisterPlugin/WorldGeneratorPlugin`.